From e39ac2967cb6a6635425c43ff74bdc50d52cc902 Mon Sep 17 00:00:00 2001
From: Adrian Moreno <amorenoz@redhat.com>
Date: Thu, 14 Nov 2019 13:23:47 +0100
Subject: [PATCH 1/2] vdpa_hack revert configure vdpa device on vring call

This is a hack to make vdpa work with virtio-user. We should probably
fix this in there instead.

Revert "vhost: configure vDPA device after set vring call message"

This reverts commit 9f90145128229b57a10bc6a37ff3c17a01b1f58f.
---
 lib/librte_vhost/vhost_user.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lib/librte_vhost/vhost_user.c b/lib/librte_vhost/vhost_user.c
index 0b72648..6595fa6 100644
--- a/lib/librte_vhost/vhost_user.c
+++ b/lib/librte_vhost/vhost_user.c
@@ -2113,7 +2113,7 @@ skip_to_post_handle:
 	vdpa_dev = rte_vdpa_get_device(did);
 	if (vdpa_dev && virtio_is_ready(dev) &&
 			!(dev->flags & VIRTIO_DEV_VDPA_CONFIGURED) &&
-			msg.request.master == VHOST_USER_SET_VRING_CALL) {
+			msg.request.master == VHOST_USER_SET_VRING_ENABLE) {
 		if (vdpa_dev->ops->dev_conf)
 			vdpa_dev->ops->dev_conf(dev->vid);
 		dev->flags |= VIRTIO_DEV_VDPA_CONFIGURED;
-- 
2.21.0

